### PR TITLE
CORE-16234: Unit tests for LedgerPersistenceRpcRequestProcessor

### PR DESCRIPTION
**Pull request by emilybowe**
Adding unit testing for the new LedgerPersistenceRpcRequestProcessor

### CODE REVIEW

**Review by mbrkic-r3 — APPROVED** [review]
LGTM

File: components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/processor/LedgerPersistenceRpcRequestProcessor.kt [hunk 1]
```diff
@@ -89,7 +90,7 @@ class LedgerPersistenceRpcRequestProcessor(
                     .record(Duration.ofNanos(System.nanoTime() - startTime))
             }
         }
-        return result as FlowEvent
+        return result.single().value as FlowEvent
```
Comment (lankydan) [inline components/ledger/ledger-persistence/src/main/kotlin/net/corda/ledger/persistence/processor/LedgerPersistenceRpcRequestProcessor.kt:93]: Why is this change needed?

**Review by lankydan — APPROVED** [review]
